assets/js/googlesitekit/data/create-notifications-store.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Provides API functions to create a datastore for notifications.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import { get } from 'googlesitekit-api';
+
+// Actions
+const ADD_NOTIFICATION = 'ADD_NOTIFICATION';
+const REMOVE_NOTIFICATION = 'REMOVE_NOTIFICATION';
+const FETCH_NOTIFICATIONS = 'FETCH_NOTIFICATIONS';
+const RECEIVE_NOTIFICATIONS = 'RECEIVE_NOTIFICATIONS';
+const RECEIVE_NOTIFICATIONS_FAILED = 'RECEIVE_NOTIFICATIONS_FAILED';
+
+/**
+ * Creates a store object that includes actions and selectors for managing notifications.
+ *
+ * The three required parameters hook up the store to the respective REST API endpoint.
+ *
+ * @since n.e.x.t
+ * @private
+ * @param {string} type       The data to access. One of 'core' or 'modules'.
+ * @param {string} identifier The data identifier, eg. a module slug like 'search-console'.
+ * @param {string} datapoint  The endpoint to request data from, e.g. 'notifications'.
+ * @return {Object} The notifications store object.
+ */
+export const createNotificationsStore = ( type, identifier, datapoint ) => {
+	const INITIAL_STATE = {
+		serverNotifications: undefined,
+		clientNotifications: {},
+		isFetchingNotifications: false,
+	};
+
+	const actions = {
+		/**
+		 * Adds a notification to the store.
+		 *
+		 * @since n.e.x.t
+		 * @param {Object} notification Notification object to add.
+		 * @return {Object} Redux-style action.
+		 */
+		addNotification( notification ) {
+			return {
+				payload: { notification },
+				type: ADD_NOTIFICATION,
+			};
+		},
+
+		/**
+		 * Removes a notification from the store.
```

Comment:
For now this seems to only remove a client-side notification; is that worth highlighting? Is there a plan to allow for the server-side notifications here to be dismissed in the future? Because right now removing a server notification based on its ID is something I'd expect to work—your point about server notifications being an implementation detail is fair, but since they are exposed like any other notification to code using the `getNotifications()` selector, there's no clarity about which kinds of notifications can be removed and which can't.

There should at least be some kind of handling for if the ID passed to `removeNotifications()` is a server-side notification that can't be removed. Even if it's just something in `console` or something—otherwise it could get quite confusing as to why a notification cannot be removed.